lib/libc/minimal/source/stdout/stdout_console.c

```diff
@@ -65,6 +65,10 @@ size_t z_impl_zephyr_fwrite(const void *_MLIBC_RESTRICT ptr, size_t size,
 		return 0;
 	}
 
+	if ((strlen(ptr) + 1) < (size * nitems)) {
+		return 0;
```

Comment:
Why are you not returning an error?

Reply:
`fwrite(3)` returns the number of items successfully written, regardless of whether there's an error, so zero is correct here.  `ferror(3)` is supposed to be used to distinguish failure modes, but minimal libc doesn't support it.